### PR TITLE
✨feat(analysis): Add trace_downstream_paths for downstream execution path tracing

### LINKED ISSUE #4: ✨ feat(Analysis): Implement Execution Path Tracing
**Summary:**

Enable tracing of all (or simple) execution paths downstream from a selected node in the dependency graph. Paths can be limited by depth or traced to a specific target node.

**Detailed Description:**

*   **Needs:** To understand the sequence of calls and the functional scope originating from a particular PL/SQL object. This helps visualize control flow and understand the breadth/depth of a feature.
*   **Requirements:** Graph traversal and path reconstruction capabilities using `networkx`.
*   **Details:** The feature should find simple paths (no repeated nodes) or all paths up to a specified depth limit. If a target node is specified, paths should lead to that target.

**Implementation Plan:**

*   **Affected Files/Objects:**
    *   `packages/dependency_analyzer/src/dependency_analyzer/analysis/analyzer.py`:
        *   The existing `find_all_paths(graph, source, target, cutoff)` can serve as a basis if a target is known.
        *   Create a new function `trace_downstream_paths(graph: nx.DiGraph, source_node: str, depth_limit: Optional[int] = None, target_node: Optional[str] = None) -> List[List[str]]`.
            *   If `target_node` is provided, use `list(nx.all_simple_paths(graph, source_node, target_node, cutoff=depth_limit))` if `depth_limit` is also provided, otherwise `list(nx.all_simple_paths(graph, source_node, target_node))`.
            *   If `target_node` is `None`:
                *   Perform a DFS/BFS traversal up to `depth_limit`.
                *   `nx.dfs_edges(graph, source_node, depth_limit=depth_limit)` can provide edges. Paths will need to be reconstructed from these edges.
                *   Alternatively, iterate through all nodes reachable within `depth_limit` (e.g., using `nx.bfs_tree(graph, source_node, depth_limit=depth_limit).nodes()`) and then find simple paths from `source_node` to each of these reachable nodes, applying the `depth_limit` to `all_simple_paths`.
            *   Careful consideration for cycles: focus on simple paths or depth-limited paths to avoid infinite loops and excessive output.
*   **Tools/Libraries:** `networkx`.
*   **Optimization:** `nx.all_simple_paths` can be computationally expensive if many paths exist. Depth limiting is crucial. For the case without a `target_node`, path reconstruction from DFS/BFS edges needs to be efficient.

**Data Requirements:**

*   **Current:** The existing graph structure (nodes and edges) is sufficient.
*   **Missing:** None for this feature.

**Pros:**

*   Visualizes control flow from a specific starting point.
*   Helps understand the breadth and depth of functionality initiated by an object.
*   Useful for debugging and understanding complex interactions.

**Cons:**

*   Can produce a very large number of paths, especially in graphs with cycles or high branching factors. This might require careful UI presentation, filtering, or further summarization.
*   Performance can be an issue for large graphs if not properly optimized (e.g., with strict depth limits).

**Migration Use Case:**

*   "Show me all sequences of PL/SQL calls that occur if `ENTRY_POINT_PROC` is executed, to understand the full logic flow we need to replicate in Java." This helps in scoping the migration of a particular feature or entry point.

**Acceptance Criteria:**

*   A new function `trace_downstream_paths` is implemented in `analyzer.py`.
*   The function correctly returns a list of paths (each path being a list of node IDs).
*   The function handles `source_node`, `target_node` (optional), and `depth_limit` (optional) parameters correctly.
*   If `target_node` is specified, only paths leading to it are returned.
*   If `depth_limit` is specified, paths do not exceed this depth.
*   The function correctly handles cases where `source_node` or `target_node` do not exist or no paths are found (returns an empty list).
*   Unit tests are added for `trace_downstream_paths`, covering various scenarios including cycles, depth limits, target nodes, and edge cases.
*   Docstrings are added for the new function.
*   Changelog is updated.

### PR DESCRIPTION
**Pull request by HrushikeshPawar**
Implements trace_downstream_paths in analyzer.py to trace all simple execution paths downstream from a selected node, with optional depth and target node limits. Adds comprehensive unit tests and updates the changelog.

- Supports depth limiting and target node filtering.
- Handles cycles and edge cases robustly.
- Includes docstrings and loguru-based logging.
- All tests pass.

closes #4

### CODE REVIEW

File: packages/dependency_analyzer/src/dependency_analyzer/analysis/analyzer.py [hunk 1]
```diff
@@ -515,6 +515,65 @@ def get_ancestors(graph: nx.DiGraph, target_node: str, depth_limit: Optional[int
...
+        if target_node:
+            # Use all_simple_paths with cutoff if provided
+            logger.debug(f"Tracing all simple paths from '{source_node}' to '{target_node}' (cutoff={depth_limit})...")
+            paths = list(nx.all_simple_paths(graph, source=source_node, target=target_node, cutoff=depth_limit))
+            logger.info(f"Found {len(paths)} paths from '{source_node}' to '{target_node}'.")
+            return paths
+        else:
+            # No target: find all simple paths from source to all reachable nodes (up to depth_limit)
+            # Use DFS to enumerate all simple paths up to depth_limit (path length = number of edges)
+            def dfs(current_path: List[str], depth: int):
```
[the remark this answers is not in the document]
Comment (HrushikeshPawar) [reply packages/dependency_analyzer/src/dependency_analyzer/analysis/analyzer.py:560]: No need

File: packages/dependency_analyzer/src/dependency_analyzer/analysis/analyzer.py [hunk 2]
```diff
@@ -673,4 +732,21 @@ def __repr__(self):
     for i, comp in enumerate(wcc):
         example_logger.info(f"  WCC {i+1}: {comp}")
 
+    # --- Test trace_downstream_paths ---
+    example_logger.info("\\n--- Testing trace_downstream_paths ---")
+    traced_paths_A = trace_downstream_paths(mock_graph, "pkg.procA", example_logger, depth_limit=2)
+    # From pkg.procA, within 2 steps: to pkg.procB, pkg.procC, standalone_func
+    # Paths: [['pkg.procA', 'pkg.procB'], ['pkg.procA', 'pkg.procC'], ['pkg.procA', 'standalone_func']]
```
[the remark this answers is not in the document]
Comment (HrushikeshPawar) [reply packages/dependency_analyzer/src/dependency_analyzer/analysis/analyzer.py:739]: Yes, the comment is wrong, bu t the functionality is correct.
```
Traced downstream paths from 'pkg.procA' (depth 2): [['pkg.procA', 'pkg.procB'], ['pkg.procA', 'pkg.procB', 'pkg.procC']]
```

File: packages/dependency_analyzer/src/dependency_analyzer/analysis/analyzer.py [hunk 3]
```diff
@@ -673,4 +732,21 @@ def __repr__(self):
...
+    example_logger.info(f"Traced downstream paths from 'pkg.procA' (depth 2): {traced_paths_A}")
+
+    traced_paths_standalone = trace_downstream_paths(mock_graph, "standalone_func", example_logger, depth_limit=2)
+    # From standalone_func, within 2 steps: to pkg.procA, util.helper
+    # Paths: [['standalone_func', 'pkg.procA'], ['standalone_func', 'util.helper']]
```
[the remark this answers is not in the document]
Comment (HrushikeshPawar) [reply packages/dependency_analyzer/src/dependency_analyzer/analysis/analyzer.py:744]: As mentioned above, comment is wrong:
```
Traced downstream paths from 'standalone_func' (depth 2): [['standalone_func', 'pkg.procA'], ['standalone_func', 'pkg.procA', 'pkg.procB'], ['standalone_func', 'util.helper'], ['standalone_func', 'util.helper', 'external.api']]
```
